cmd/finch-daemon/main.go

```diff
@@ -84,14 +110,10 @@ func run(options *DaemonOptions) error {
 	serverWg := &sync.WaitGroup{}
 	serverWg.Add(1)
 
-	listener, err := net.Listen("unix", options.socketAddr)
+	var listener net.Listener
+	listener, err = getListener(options)
```

Comment:
nit: can we get rid of the explicit declaration and use `:=` in the next line?

```suggestion
	listener, err := getListener(options)
```